boxsdk/auth/oauth2.py

```diff
@@ -240,17 +242,40 @@ def send_token_request(self, data, access_token, expect_refresh_token=True):
             url,
             data=data,
             headers=headers,
-            access_token=access_token
+            access_token=access_token,
         )
         if not network_response.ok:
             raise BoxOAuthException(network_response.status_code, network_response.content, url, 'POST')
         try:
             response = network_response.json()
-            self._access_token = response['access_token']
-            self._refresh_token = response.get('refresh_token', None)
-            if self._refresh_token is None and expect_refresh_token:
+            access_token = response['access_token']
+            refresh_token = response.get('refresh_token', None)
+            if refresh_token is None and expect_refresh_token:
                 raise BoxOAuthException(network_response.status_code, network_response.content, url, 'POST')
         except (ValueError, KeyError):
             raise BoxOAuthException(network_response.status_code, network_response.content, url, 'POST')
-        self._store_tokens(self._access_token, self._refresh_token)
+        self._store_tokens(access_token, refresh_token)
         return self._access_token, self._refresh_token
+
+    def revoke(self):
+        """
+        Revoke the authorization for the current access/refresh token pair.
+        """
+        with self._refresh_lock:
+            token_to_revoke, _ = self._get_tokens()
+            if token_to_revoke is None:
```

Comment:
If we did something like

```python
access_token, refresh_token = self._get_tokens()
token_to_revoke = access_token or refresh_token
```

Then we can support use cases where the user only stored the refresh token, not the access token, and now wants to revoke it.